<a href="https://colab.research.google.com/github/Shimanto-125/flyrank_internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# Name of my lane: Refresh / Content Opportunity Scoring<br>

## Reason of selecting the lane:
I selected the lane because it focuses on identifying which content pages should be reviewed first based on observable search and engagement signals. This lane aligns with the provided starter dataset and supports a real business decision by helping content teams prioritize pages that may benefit from refreshing. Rather than trying to automate SEO decisions, the goal is to build a transparent decision-support system that ranks pages for human review.

In [14]:
print("Lane selected: Refresh / Content Opportunity Scoring")

Lane selected: Refresh / Content Opportunity Scoring


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research question:
Which content pages should be reviewed first for refresh based on search performance and engagement signals?
<br>
###Decision:
The project improves the decision of which content pages should be prioritized for review and refresh.<br>

###Action:
The SEO or Content team will review the highest-ranked pages and decide whether to update titles, improve content, expand information, or monitor performance.
<br>
###Unit of Analysis:
One row represents one content page.
<br>
###Output:
A ranked list of content pages with refresh priority scores and supporting evidence.<br>
###Cost of a Wrong Recommendation:
If the recommendation is incorrect, the team may spend time and resources updating low-impact pages while important pages continue losing visibility and traffic. This reduces productivity and may delay improvements for high-value content.

In [15]:
print("Research question defined.")

Research question defined.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")
print(f"Total pages: {len(df)}")
df.head(7)


Total pages: 30000


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7
5,content_d4084a4bc775,client_f369cb89fc,720.0,1.00,HIGH,1.05,keyword article,transactional,3080.0,18178.0,...,15000-25000,0.03,8.5,0.00,25.00,0.0,good,page_1,down,-38.9
6,content_9a34b442b552,client_8722616204,0.0,0.00,LOW,0.00,keyword article,informational,3059.0,20810.0,...,15000-25000,0.00,7.0,0.00,0.00,0.0,low,page_1,down,-92.3


In [17]:
print("Average CTR:", round(df["ctr"].mean(), 4))
print("Average Position:", round(df["avg_position"].mean(), 2))
print("---------------------------------------------------")

print("Content Type Distribution")
print(df["content_type"].value_counts())

Average CTR: 0.5107
Average Position: 16.34
---------------------------------------------------
Content Type Distribution
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64


###The starter dataset was explored to understand whether this lane is suitable.<br>

The analysis shows:

* Total number of content pages: $30000$
* Average CTR: $0.5107$
* Average search position: $16.34$<br>

These observations suggest that the dataset contains measurable search and engagement signals that can support ranking pages for refresh opportunities. The variation across pages indicates that prioritization could help teams allocate their review effort more effectively.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project provides decision-support based on observed search and engagement signals.<br>

The ranking identifies pages that may deserve review because they show patterns associated with lower performance or higher opportunity.<br>

This work does not prove that refreshing a page will increase traffic, improve rankings, or change Google's search algorithm. The results are observational and intended to support human decision-making rather than replace it.<br>

## Self-check

Before you submit, confirm each line honestly:

- [ x] Every section above is filled — markdown thinking AND the code that backs it
- [x ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x ] No client names, URLs, or private queries anywhere
- [ x] My claims use careful words: observed, measured, directional, decision-support
- [x ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.